lib/puppet/functions.rb

```diff
@@ -82,25 +82,33 @@
 # be the same as the number of parameters, and all of the parameters are of
 # type 'Any'.
 #
-# To express that the last parameter captures the rest, the method
-# `last_captures_rest` can be called. This indicates that the last parameter is
-# a varargs parameter and will be passed to the implementing method as an array
-# of the given type.
+# The following methods can be used for declaring a parameter
 #
-# When defining a dispatch for a function, the resulting dispatch matches
-# against the specified argument types and min/max occurrence of optional
-# entries. When the dispatch makes the call to the implementation method the
-# arguments are simply passed and it is the responsibility of the method's
-# implementor to ensure it can handle those arguments (i.e. there is no check
-# that what was declared as optional actually has a default value, and that
-# a "captures rest" is declared using a `*`).
+#  - _param_ - the parameter must be given in a call.
+#  - _optional_param_ - the parameter may be missing in the call. May not be used on a parameter that is followed by a required parameter
+#  - _repeated_param_ - the type specifies a repeating type that occurs 0 to "infinite" number of times. It may only appear last or just before a block parameter.
+#  - _block_param_ - a block must be given in the call. May only appear last.
+#  - _optional_block_param_ - a block may be given in the call. May only appear last.
 #
-# @example Varargs
+# The method name _required_param_ is an alias for _param_ and _required_block_param_ is an alias for _block_param_
+#
+# A parameter declaration takes 2 arguments:
+#  - _type_ A string that must conform to a type in the puppet language
+#  - _name_ A symbol denoting the parameter name
+#
+# Both arguments are optional when declaring a block parameter. The _type_ defaults to "Callable"
+# and the _name_ to :block.
+#
+# When defining a dispatch for a function, the resulting dispatch matches against the specified
+# argument types. When the dispatch makes the call to  the implementation method the arguments are
```

Comment:
"Note that the dispatch definition is used to match arguments given in a call to the function with the defined parameters. It then dispatches the call to the implementation method simply passing the given arguments on to that method without any further processing and it is the responsibility of that method's implementor to ensure that it can handle those arguments."